### PR TITLE
First version of settings UI

### PR DESCRIPTION
**Pull request by susnux**
This is a first version of the UI for https://github.com/nextcloud/server/issues/40160#issuecomment-1774393690

The styling is quite rough and needs to be adjusted in a follow up.

### CODE REVIEW

File: lib/Settings/Admin.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+	/**
+	 * @return int whether the form should be rather on the top or bottom of
+	 * the admin section. The forms are arranged in ascending order of the
+	 * priority values. It is required to return a value between 0 and 100.
+	 *
+	 * E.g.: 70
+	 */
+	public function getPriority(): int {
+		return 0;
```
Comment (szaimen) [inline lib/Settings/Admin.php:62]: Can this be set to something higher than 0 so that AIO can still show its section above it?

See https://github.com/nextcloud/all-in-one/blob/b2ca0b0e22b5c35fcd1eb629726ccd86abdafa29/app/lib/Settings/Admin.php#L80-L82

Comment (szaimen) [reply lib/Settings/Admin.php:62]: I think this needs to stay the first section otherwise people will miss it because they need to scroll down:
image

Comment (szaimen) [reply lib/Settings/Admin.php:62]: Actually in my opinion the settings UI that you are working on should even be shown below the `Security & setup warnings` section but this is less important than being able to show the Nextcloud All-in-one section on top.

Comment (susnux) [reply lib/Settings/Admin.php:62]: I think it was decided it should be the very top one to allow admins to see issues without the need to scroll.
Any other options, including AIO are just settings and so you know you want to go there and can scroll down.

But this are status checks that show unexpected status information, so you should see any warning or error direct without need to scroll down.

Comment (susnux) [reply lib/Settings/Admin.php:62]: (I think the AIO settings should even be below the setup warnings as it does not provide any important information but only provides a configuration which admins will anyways look for)

Comment (susnux) [reply lib/Settings/Admin.php]: Done :+1:

File: lib/Settings/Admin.php [hunk 2]
```diff
@@ -59,7 +59,8 @@ public function getSection(): string {
 	 * E.g.: 70
 	 */
 	public function getPriority(): int {
-		return 0;
+		// We should be above all other settings but allow edge cases like AiO to override
+		return 1;
```
Comment (szaimen) [inline lib/Settings/Admin.php:63]: Thanks, LGTM! :)

**Review by szaimen — COMMENTED** [review]
I just tested this with nextcloud-easy-test 
```
docker run -it \
--name nextcloud-easy-test \
-p 8443:443 \
-e TRUSTED_DOMAIN=192.168.24.128 \
--volume="nextcloud_easy_test_npm_cache_volume:/var/www/.npm" \
-e SERVER_BRANCH=feat/improve-setup-checks-wording \
-e ADMINOVERVIEW_BRANCH=feat/initial-ui \
-e NEXTCLOUD_LOGLEVEL=0 \
ghcr.io/szaimen/nextcloud-easy-test:latest
```
And the initial loading of the admin overview (log in the first time and click on Administration settings) took like 20-30s  while it was not switching views and only showing a spinner in the dropdown. Not sure how or if we can fix this but this is not acceptable imho?

Here a video of this:

https://github.com/nextcloud/admin_overview/assets/42591237/a7eb053a-7f5d-4916-b467-481e72dff805

**Comment by susnux** [comment]
yes the status loading should be moved to some API

**Comment by come-nc** [comment]
@szaimen Does your test instance have internet access?
There is an internet access setup check with a timeout of 30s which is way too slow when internet connectivity is not working. I plan on adding at least a cache for it.

**Comment by szaimen** [comment]
> There is an internet access setup check with a timeout of 30s which is way too slow when internet connectivity is not working. I plan on adding at least a cache for it.

yes it has internet connection... maybe the problem is that it does not have redis and only sqllite?

**Comment by come-nc** [comment]
> > There is an internet access setup check with a timeout of 30s which is way too slow when internet connectivity is not working. I plan on adding at least a cache for it.
> 
> yes it has internet connection... maybe the problem is that it does not have redis and only sqllite?

Can you enable debug log and look for "Running check " debug log entries, and see if some of them have a long time between them?

**Comment by szaimen** [comment]
> Can you enable debug log and look for "Running check " debug log entries, and see if some of them have a long time between them?

So I got the following output which indeed points at the internet check (Apparently the internet in the test instance is indeed very slow, something like 500Kbit/s:
```
[Thu Oct 26 09:24:10.205969 2023] [php:notice] [pid 667] [client 192.168.24.1:53891] [nextcloud][no app in context][0] {"reqId":"silJUk0iSIYIjg4qkaWn","level":0,"time":"2023-10-26T09:24:10+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/admin/overview","message":"Running check OCA\\\\DAV\\\\SetupChecks\\\\NeedsSystemAddressBookSync","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:24:10.207526 2023] [php:notice] [pid 667] [client 192.168.24.1:53891] [nextcloud][no app in context][0] {"reqId":"silJUk0iSIYIjg4qkaWn","level":0,"time":"2023-10-26T09:24:10+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/admin/overview","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\CheckUserCertificates","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:24:10.208078 2023] [php:notice] [pid 667] [client 192.168.24.1:53891] [nextcloud][no app in context][0] {"reqId":"silJUk0iSIYIjg4qkaWn","level":0,"time":"2023-10-26T09:24:10+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/admin/overview","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\DefaultPhoneRegionSet","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:24:10.209028 2023] [php:notice] [pid 667] [client 192.168.24.1:53891] [nextcloud][no app in context][0] {"reqId":"silJUk0iSIYIjg4qkaWn","level":0,"time":"2023-10-26T09:24:10+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/admin/overview","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\InternetConnectivity","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:25:01.815133 2023] [php:notice] [pid 667] [client 192.168.24.1:53891] [nextcloud][no app in context][0] {"reqId":"silJUk0iSIYIjg4qkaWn","level":0,"time":"2023-10-26T09:25:01+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/admin/overview","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\LegacySSEKeyFormat","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:25:01.815538 2023] [php:notice] [pid 667] [client 192.168.24.1:53891] [nextcloud][no app in context][0] {"reqId":"silJUk0iSIYIjg4qkaWn","level":0,"time":"2023-10-26T09:25:01+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/admin/overview","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\PhpDefaultCharset","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:25:01.815983 2023] [php:notice] [pid 667] [client 192.168.24.1:53891] [nextcloud][no app in context][0] {"reqId":"silJUk0iSIYIjg4qkaWn","level":0,"time":"2023-10-26T09:25:01+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/admin/overview","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\PhpOutdated","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:25:01.816329 2023] [php:notice] [pid 667] [client 192.168.24.1:53891] [nextcloud][no app in context][0] {"reqId":"silJUk0iSIYIjg4qkaWn","level":0,"time":"2023-10-26T09:25:01+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/admin/overview","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\PhpOutputBuffering","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:25:01.816751 2023] [php:notice] [pid 667] [client 192.168.24.1:53891] [nextcloud][no app in context][0] {"reqId":"silJUk0iSIYIjg4qkaWn","level":0,"time":"2023-10-26T09:25:01+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/admin/overview","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\ReadOnlyConfig","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:25:01.817340 2023] [php:notice] [pid 667] [client 192.168.24.1:53891] [nextcloud][no app in context][0] {"reqId":"silJUk0iSIYIjg4qkaWn","level":0,"time":"2023-10-26T09:25:01+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/admin/overview","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\SupportedDatabase","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:25:03.099694 2023] [php:notice] [pid 668] [client 192.168.24.1:53941] [nextcloud][no app in context][0] {"reqId":"wDo7XeK9WPlZOqakKlpa","level":0,"time":"2023-10-26T09:25:03+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/ajax/checksetup","message":"Running check OCA\\\\DAV\\\\SetupChecks\\\\NeedsSystemAddressBookSync","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:25:03.100607 2023] [php:notice] [pid 668] [client 192.168.24.1:53941] [nextcloud][no app in context][0] {"reqId":"wDo7XeK9WPlZOqakKlpa","level":0,"time":"2023-10-26T09:25:03+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/ajax/checksetup","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\CheckUserCertificates","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:25:03.100833 2023] [php:notice] [pid 668] [client 192.168.24.1:53941] [nextcloud][no app in context][0] {"reqId":"wDo7XeK9WPlZOqakKlpa","level":0,"time":"2023-10-26T09:25:03+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/ajax/checksetup","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\DefaultPhoneRegionSet","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}
[Thu Oct 26 09:25:03.101005 2023] [php:notice] [pid 668] [client 192.168.24.1:53941] [nextcloud][no app in context][0] {"reqId":"wDo7XeK9WPlZOqakKlpa","level":0,"time":"2023-10-26T09:25:03+00:00","remoteAddr":"192.168.24.1","user":"admin","app":"no app in context","method":"GET","url":"/index.php/settings/ajax/checksetup","message":"Running check OCA\\\\Settings\\\\SetupChecks\\\\InternetConnectivity","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36","version":"28.0.0.2","data":[]}

```

**Comment by come-nc** [comment]
> So I got the following output which indeed points at the internet check (Apparently the internet in the test instance is indeed very slow, something like 500Kbit/s:

Note that it also happens when nextcloud.com is offline :see_no_evil:
